controllers/operatorpolicy_status.go

```diff
@@ -282,6 +282,21 @@ func calculateComplianceCondition(policy *policyv1beta1.OperatorPolicy) metav1.C
 		}
 	}
 
+	idx, cond = policy.Status.GetCondition(deprecationType)
+	if !policy.Spec.ComplianceType.IsMustNotHave() &&
+		policy.Spec.ComplianceConfig.DeprecationAvailable == "Compliant" {
```

Comment:
Other conditions change the `Status` of the condition based on the complianceConfig, but I feel like your approach here is better.

If you follow my other suggestion, I think this part would need to be updated like this:
```suggestion
		policy.Spec.ComplianceConfig.DeprecationsPresent == "NonCompliant" {
```